File: a6/a6.py
import string
class InvalidKeyException(Exception):
	pass
class InvalidMode(Exception):
	pass
  
class Cipher:
  
	@staticmethod
	def numeric_key(st):
		st =st.strip().lower()
		return [ord(x)-ord('a') for x in list(st)]
    
	@staticmethod
	def caesar(st,key,mode='encrypt'):
		st= st.lower()
		if mode not in ['encrypt','decrypt']:
			raise InvalidMode("Mode must be encrypt or decrypt")
    		if key <0 or key > 25:
      			raise InvalidKeyException("Key must be between 0 and 25")
    		return st.translate(Cipher.table(key,mode))
  
  	@staticmethod
  	def table(key,mode):
  		assert (mode in ['encrypt','decrypt']),"Invalid Mode"
    		alpha = string.ascii_lowercase
    		salpha = alpha[key:]+alpha[:key]
    		if mode == 'encrypt':
      			return string.maketrans(alpha,salpha)
    		return string.maketrans(salpha,alpha)
    
  	@staticmethod
  	def vignere(st,keystr,mode='encrypt'):
  		st = st.strip().lower()
    		l = len(st)
    		keys = Cipher.numeric_key(keystr)
    		tables = [Cipher.table(key,mode) for key in keys]
    		return "".join([c.translate(tables[i%l]) for i,c in enumerate(st)])
    
pt=raw_input("Enter Plain Text: ")
k1=raw_input("Enter key(Interger between 0 to 26): ")
k2=raw_input("Enter key(Size should same as Plain Text): ")

print "------------Ceasar Cipher------------"
cipher =  Cipher.caesar(pt,int(k1)+1,mode='encrypt')
print 'Plain Text: ', Cipher.caesar(cipher,int(k1)+1,mode='decrypt')
print 'Cipher Text: ', cipher

print "------------Vignere Cipher------------"
ct = Cipher.vignere(pt,str(k2),mode='encrypt')
print 'Plain Text: ', Cipher.vignere(ct,str(k2),mode='decrypt')
print 'Cipher Text: ', ct

File: a6/a6.cpp
#include<bits/stdc++.h>
using namespace std;
class encrpyt
{
	public : 
	void getinput();
	void output(string);
	void ceasar_cipher(string, int);
	void vignere_cipher(string, string);
};

void encrpyt :: getinput()
{
	cout<<"1 to for Ceasar Cipher 		2 for Vignere Cipher"<<endl;
	int ch;
	cin>>ch;
	if (ch == 1)
	{
		cout<<"Enter the plain text"<<endl;
		string text;
		cin>>text;
		cout<<"Enter the key between 0 to 25"<<endl;
		int key;
		cin>>key;
		ceasar_cipher(text,key);
	}
	else
	{
		cout<<"Enter the plain text" <<endl;
		string text;
		cin>>text;
		cout<<"Enter the key"<<endl;
		string key;
		cin>>key;
		vignere_cipher(text,key);

	}
}

void encrpyt :: ceasar_cipher(string text, int key)
{
	string result = "";

	for (int i=0;i<text.length();i++)
    	{
        	if (isupper(text[i]))
   			result += char(int(text[i]+key-'A')%26 +'A');
       		else
   		     	result += char(int(text[i]+key-'a')%26 +'a');
   	}
    	output(result);
}

void encrpyt :: vignere_cipher (string text, string key)
{
	int x = text.length();

	for(int i=0 ; ;i++)
	{	
		if(x == i)
			i = 0;
		if(key.size()== text.length())
			break;
		key.push_back(key[i]);
	}
	
	string cipher_text = "";

	for (int i = 0 ; i<text.length(); i++)
	{
		int c;		
		if (isupper(text[i]))
			c = (text[i]+key[i]-(2*'A'))%26+'A';
		else 
			c = (text[i]+key[i]-(2*'a'))%26+'a';
			cout<<c;
		cipher_text.push_back(c);
	}
	output(cipher_text);
}

void encrpyt :: output(string pass)
{
	cout<<"The encrpyted text is  "<<pass<<endl;
}

int main()
{
	encrpyt e;
	e.getinput();
	return 0;
}
